library/ceph_pool.py

```diff
@@ -0,0 +1,121 @@
+#!/usr/bin/python
+#coding: utf-8 -*-
+
+DOCUMENTATION = """
+---
+author: Michael Sambol
+module: ceph_pool
+short_description: Creates ceph pool and ensures correct pg count
+description:
+  There are three possible outcomes:
+    1/ Create a new pool if it doesn't exist
+    2/ Set the pool's pg count to correct number
+    3/ Nothing: pool is created and pg count is correct
+options:
+  pool_name:
+    description:
+      - The pool in question: create it or ensure correct pg count
+    required: true
+  constant:
+    description:
+      - A constant multiplier used in calculating pg count
+    required: true
+  copies:
+    description:
+      - The number of copies the cluster keeps
+    required: true
+"""
+
+EXAMPLES = """
+# ceph_pool can only be run on nodes that have an admin keyring
+# pool_name = default, constant = 100, copies = 3
+- ceph_pool:
+    pool_name: default
+    constant: 100
+    copies: 3
+  register: pool_output
+  run_once: true
+  delegate_to: "{{ groups[ceph.monitor_group_name][0] }}"
+"""
+
+import time
+
+
+def increase_pg_count(module, osds, pool_name,
+                      current_pg_count, desired_pg_count):
+    max_increase_per_pass = osds * 32
+    diff = desired_pg_count - current_pg_count
+    while diff > 0:
+        if diff <= max_increase_per_pass:
+            cmd = ['ceph', 'osd', 'pool', 'set', pool_name,
+                   'pg_num', str(desired_pg_count)]
+            rc, out, err = module.run_command(cmd, check_rc=True)
+            time.sleep(30)
+            cmd = ['ceph', 'osd', 'pool', 'set', pool_name,
+                   'pgp_num', str(desired_pg_count)]
+            rc, out, err = module.run_command(cmd, check_rc=True)
+            diff = 0
+        else:
+            current_pg_count = current_pg_count + max_increase_per_pass
+            cmd = ['ceph', 'osd', 'pool', 'set', pool_name,
+                   'pg_num', str(current_pg_count)]
+            rc, out, err = module.run_command(cmd, check_rc=True)
+            time.sleep(30)
+            cmd = ['ceph', 'osd', 'pool', 'set', pool_name,
+                   'pgp_num', str(current_pg_count)]
+            rc, out, err = module.run_command(cmd, check_rc=True)
+            diff = diff - max_increase_per_pass
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            pool_name=dict(required=True),
+            constant=dict(required=True),
+            copies=dict(required=True),
+        ),
+    )
+
+    pool_name = module.params.get('pool_name')
+    constant = int(module.params.get('constant'))
+    copies = int(module.params.get('copies'))
+
+    # determine number of osds in cluster
+    cmd = ['ceph', 'osd', 'stat']
+    rc, out, err = module.run_command(cmd, check_rc=True)
+    osds = int(out.splitlines()[0].split(":")[1].strip().split(" ")[0])
```

Comment:
That's a lot of text manipulation. It'd help if you put a comment in with the expected input string and the expected output string, which will help should the command output ever change.